pkg/controllers/vdb/podfacts.go

```diff
@@ -1318,3 +1370,12 @@ func (p *PodFacts) FindSecondarySubclustersWithDifferentImage() (scs []string, p
 	}
 	return scs, priScImage
 }
+
+func isSubclusterShutdown(vdb *vapi.VerticaDB, sc *vapi.Subcluster) bool {
+	sbName := vdb.GetSubclusterSandboxName(sc.Name)
+	sb := vdb.GetSandbox(sbName)
+	if sb != nil {
+		return sb.Shutdown
+	}
+	return sc.Shutdown
```

Comment:
If sb == nil, we should return false since the subcluster isn't in a sandbox.

Reply:
Later we are going to support `stop subcluster` and it won't be limited to sandboxes. So it will be possible for a main cluster's subcluster to be "shutdown".

Reply:
Because we want the pfacts to reflect the reality as soon as possible. The moment the user set `spec.sandboxes[].shutdown` to true, we no longer allow the operator to restart those pods. So if a reconciler collects the pfacts before the operator updates the subclusters, the will see those pods with the correct shutdown state. The sandbox owns its subcluster.
I know about the limitation( if you stop_sc in a sandbox you will get false although subcluster shutdown state is true. This PR is for stop_sandbox so I was not thinking too much about `stop_sc`).  Here is the change I can make:
```
func (p *PodFacts) isSubclusterShutdown(vdb *vapi.VerticaDB, sc *vapi.Subcluster) bool {
        if p.SandboxName == vapi.MainCluster {
                return sc.Shutdown
        }
	sb := vdb.GetSandbox(p.SandboxName)
	if sb != nil {
                 if sb.Shutdown {
                         return true
                 }
	}
	return sc.Shutdown
}
```
In case of `stop_sandbox` getting the info from the sandbox is faster.